src/plugins/histogram/index.js

```diff
@@ -0,0 +1,215 @@
+import videojs from 'video.js';
+
+import './histogram.scss';
+
+// Default options for the plugin.
+let defaults = {};
+
+/**
+ * Function to invoke when the player is ready.
+ *
+ * @function onPlayerReady
+ * @param    {Player} player
+ *           A Video.js player object.
+ *
+ * @param    {Object} [options={}]
+ *           A plain object containing options for the plugin.
+ */
+const onPlayerReady = function onPlayerReady(player, options) {
+  player.addClass('vjs-histogram');
+  player.histogram = new HistogramPlugin(player, options);
+};
+
+/**
+ * A video.js plugin.
+ *
+ * In the plugin function, the value of `this` is a video.js `Player`
+ * instance. You cannot rely on the player being in a "ready" state here,
+ * depending on how the plugin is invoked. This may or may not be important
+ * to you; if not, remove the wait for "ready"!
+ *
+ * @function histogram
+ * @param    {Object} [options={}]
+ *           An object of options left to the plugin author to define.
+ */
+const histogram = function histogram(options) {
+  let _this = this;
+
+  this.ready(function () {
+    onPlayerReady(_this, videojs.mergeOptions(defaults, options));
+  });
+};
+
+/**
+ * Histogram class.
+ *
+ * This class performs all functions related to displaying the Histogram bar.
+ */
+const HistogramPlugin = (function () {
+
+  /**
+   * Plugin class constructor, called by videojs on
+   * ready event.
+   *
+   * @function  constructor
+   * @param    {Player} player
+   *           A Video.js player object.
+   *
+   * @param    {Object} [options={}]
+   *           A plain object containing options for the plugin.
+   */
+  function HistogramPlugin(player, options) {
+    this.player = player;
+    this.options = options;
+    this.listenForDurationChange();
+    this.initializeHistogram();
+    this.registeredEvents = {};
+    return this;
+  }
+
+  HistogramPlugin.prototype.src = function src(source) {
+    this.resetPlugin();
+    this.options.src = source;
+    this.initializeHistogram();
+  };
+
+  HistogramPlugin.prototype.detach = function detach() {
+    this.resetPlugin();
+  };
+
+  HistogramPlugin.prototype.resetPlugin = function resetPlugin() {
+    if (this.histogramHolder) {
+      this.histogramHolder.parentNode.removeChild(this.histogramHolder);
+    }
+    delete this.progressBar;
+    delete this.histogramHolder;
+    delete this.lastStyle;
+  };
+
+  HistogramPlugin.prototype.listenForDurationChange = function listenForDurationChange() {
+    this.player.on('durationchange', function () {
+      // ToDo
+    });
+  };
+
+  /**
+   * Bootstrap the plugin.
+   */
+
+  HistogramPlugin.prototype.initializeHistogram = function initializeHistogram() {
+    let _this2 = this;
+
+    if (!this.options.src) {
+      return;
+    }
+
+    this.getHistogramFile(this.options.src).then(function (res) {
+      _this2.setupHistogramElement();
+      if (_this2.histogramHolder) {
+        _this2.createHistogram(res);
+      }
+    });
```

Comment:
You need this2 because of the dynamic binding of `this` keyword  that happens in a regular function, if you use an arrow function here you can get rid of the confusing `this2` declaration and just use regular `this` that will be lexically bound, which i think will work.
```suggestion
    this.getHistogramFile(this.options.src).then(res => {
      this.setupHistogramElement();
      if (this.histogramHolder) {
        this.createHistogram(res);
      }
    });
```